leaflet-treering.js

```diff
@@ -348,67 +348,149 @@ function MeasurementData (dataObject) {
+    // distance: point[i + 1] to inserted point
+    var dis_plus_to_insert = distanceCalc(pt_i_plus, pt_insert);
+    // distance: point[i - 1] to inserted point
+    var dis_minus_to_insert = distanceCalc(pt_i_minus, pt_insert);
+
+    /* Law of cosines:
+       * c = distance between inserted point and points[i + 1] or points[i - 1]
+       * b = distance between points[i] and points[i + 1] or points[i - 1]
+       * a = distance between inserted points and points[i]
+       Purpose is to find angle C for triangles formed:
+       * Triangle [i + 1] = points[i], points[i + 1], inserted point
+       * Triangle [i - 1] = points[i], points[i - 1], inserted point
+       Based off diagram from: https://www2.clarku.edu/faculty/djoyce/trig/formulas.html#:~:text=The%20law%20of%20cosines%20generalizes,cosine%20of%20the%20opposite%20angle.
+    */
+    // numerator and denominator for calculating angle C using Law of cosines (rearranged original equation)
+    var numeratorPlus = (dis_plus_to_insert ** 2) - ((dis_i_to_insert ** 2) + (dis_i_to_plus ** 2));
+    var denominatorPlus = -2 * dis_i_to_insert * dis_i_to_plus;
+    var numeratorMinus = (dis_minus_to_insert ** 2) - ((dis_i_to_insert ** 2) + (dis_i_to_minus ** 2));
+    var denominatorMinus = -2 * dis_i_to_insert * dis_i_to_minus;
+    var anglePlus = Math.acos(numeratorPlus/denominatorPlus);
+    var angleMinus = Math.acos(numeratorMinus/denominatorMinus);
+
+    // smaller angle determines connecting lines
+    if (anglePlus < angleMinus) {
+      i++;
+    };
+
     var new_points = this.points;
-    var second_points = Object.values(this.points).splice(i, this.index - 1);
+    var second_points = this.points.slice().splice(i, this.index - 1);
     var k = i;
-    var year_adjusted = this.points[i].year;
+    var year_adjusted;
     var earlywood_adjusted = true;
 
-    if (this.points[i - 1].earlywood && hasLatewood) {
-      year_adjusted = this.points[i - 1].year;
-      earlywood_adjusted = false;
-    } else if (this.points[i - 1].start) {
-      year_adjusted = this.points[i + 1].year;
+    if (this.points[i - 1]) {
+      if (this.points[i - 1].earlywood && hasLatewood) {
+        year_adjusted = this.points[i - 1].year;
+        earlywood_adjusted = false;
+      } else if (this.points[i - 1].start) {
+        year_adjusted = this.points[i + 1].year;
+          if (this.points[i - 2] && this.points[i - 2].earlywood && hasLatewood) {
+            earlywood_adjusted = false;
+          };
+      } else {
+        year_adjusted = this.points[i - 1].year + 1;
+      };
     } else {
-      year_adjusted = this.points[i - 1].year + 1;
-    }
-    new_points[k] = {'start': false, 'skip': false, 'break': false,
-      'year': year_adjusted, 'earlywood': earlywood_adjusted,
-      'latLng': latLng};
+      alert('Please insert new point closer to connecting line.')
+    };
 
-    var tempK = k;
+    if (Number.isInteger(year_adjusted)) {
```

Comment:
should we do anything at all if year_adjusted ins't an int? And is there  a chance it'll be something other than null, but not an int?  Would it be safe to do 
if(!year_adjusted) {
return;
}

The more you can avoid having big chunks of code wrapped in a conditional, the better, as it makes things easier to follow. So, if the "else" condition for this case is "aw crap something went wrong" just check for the failure and bail out, instead of wrapping all the code in the safety check.